portfolio/src/main/webapp/gallery.js

```diff
@@ -0,0 +1,48 @@
+
+var travelSlideshow = {
+  slideIndex: 0,
+  plusSlides: function(offset) {
+    this.showSlides(this.slideIndex += offset);
+  }, 
+  currentSlide: function(newSlideIndex) {
+    this.showSlides(this.slideIndex = newSlideIndex);
+  },
+  hideAllSlides: function(slides) {
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";  
+    }
+  },
+  disableAllDots: function(dots) {
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+  },
+  showSlides: function(activatedSlide) {
+    var i;
```

Comment:
What's the variable i being used for? I think these need to go into the for loops above.

I would also recommend to declare variables as close to where they are used.